pkg/downloader/downloader.go

```diff
@@ -239,18 +240,27 @@ func Download(ctx context.Context, local, remote string, opts ...Opt) (*Result,
 				return nil, err
 			}
 		} else {
-			if err := copyLocal(ctx, localPath, shadData, ext, o.decompress, o.description, o.expectedDigest); err != nil {
-				return nil, err
+			if match, err := matchLastModified(ctx, shadTime, remote); err != nil {
+				logrus.WithError(err).Info("Failed to retrieve last-modified for cached digest-less image; using cached image.")
+			} else if match {
+				if err := copyLocal(ctx, localPath, shadData, ext, o.decompress, o.description, o.expectedDigest); err != nil {
+					return nil, err
+				}
+			} else {
+				logrus.Info("Re-downloading digest-less image: last-modified mismatch detected.")
```

Comment:
Maybe print the last-modified value here

Reply:
I think if you print the last-modified time, then it would be nice to print both the time for the cached version, and for the remote version.

Reply:
I rewrote `matchLastModified` in the process of including the actual `last-modified` timestamp in the logs, so I would appreciate it if you could review that part again.